addon/loaders/js.js

```diff
@@ -15,11 +15,17 @@ export default nodeLoader(function js(uri) {
       return resolve();
     }
 
-    const script = createLoadElement('script', resolve, reject);
+    Ember.run.schedule('afterRender', () => {
+      try {
+        const script = createLoadElement('script', resolve, reject);
 
-    script.src = uri;
-    script.async = false;
+        script.src = uri;
+        script.async = false;
 
-    document.head.appendChild(script);
+        document.head.appendChild(script);
+      } catch(e) {
+        reject(e);
```

Comment:
This also maintains existing behavior. Currently if any of those lines error, the promise constructor will catch them.  As this PR moves them to be async, we must manage that ourselves.